src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java

```diff
@@ -144,7 +144,25 @@ public record VectorizeConfig(
                   type = SchemaType.OBJECT)
               @JsonProperty("parameters")
               @JsonInclude(JsonInclude.Include.NON_NULL)
-              Map<String, Object> parameters) {}
+              Map<String, Object> parameters) {
+
+        public VectorizeConfig(
+            String provider,
+            String modelName,
+            Map<String, String> authentication,
+            Map<String, Object> parameters) {
+          this.provider = provider;
+          // HuggingfaceDedicated does not need user to specify model
+          // use endpoint-defined-model as placeholder
+          if (provider.equals(ProviderConstants.HUGGINGFACE_DEDICATED)) {
+            this.modelName = "endpoint-defined-model";
```

Comment:
Should this be extracted into constant in provider constants as well?

Reply:
good point!